src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectSystem/RazorProjectHostBase.cs

```diff
@@ -159,8 +170,19 @@ private ProjectSnapshotManagerBase GetProjectManager()
 
         protected async Task UpdateAsync(Action action)
         {
-            await CommonServices.ThreadingService.SwitchToUIThread();
-            action();
+            // The passed-in action is intended to execute solely on our specialized single-threaded
+            // foreground scheduler. However, there are times when the action may need to temporarily
+            // switch to the UI thread to perform a certain task. We want to make sure that no other
```

Comment:
Hmmm, not sure I follow. Could you provide an example? Or if chatting offline is easier we can do that too 😄